core/src/config/errors.ts

```diff
@@ -16,48 +26,57 @@ export const DEFAULT = {
           websocket: GrouparooErrorSerializer,
           specHelper: GrouparooErrorSerializer,
         },
+        // See ActionProcessor#applyDefaultErrorLogLineFormat to see an example of how to customize
+        actionProcessor:
+          null as ActionProcessor<any>["applyDefaultErrorLogLineFormat"],
       },
 
       // ///////////
       // ACTIONS //
       // ///////////
 
       // When a params for an action is invalid
-      invalidParams: (data, validationErrors: string[]) => {
+      invalidParams: (
+        data: ActionProcessor<any>,
+        validationErrors: Array<string | Error>
+      ) => {
         if (validationErrors.length >= 0) return validationErrors[0];
         return "validation error";
       },
 
       // When a required param for an action is not provided
-      missingParams: (data, missingParams: string[]) => {
+      missingParams: (data: ActionProcessor<any>, missingParams: string[]) => {
         return `${missingParams[0]} is a required parameter for this action`;
       },
 
       // user requested an unknown action
-      unknownAction: (data) => {
+      unknownAction: (data: ActionProcessor<any>) => {
         return `unknown action or invalid apiVersion`;
       },
 
       // action not useable by this client/server type
-      unsupportedServerType: (data) => {
+      unsupportedServerType: (data: ActionProcessor<any>) => {
         return `this action does not support the ${data.connection.type} connection type`;
       },
 
       // action failed because server is mid-shutdown
-      serverShuttingDown: (data) => {
+      serverShuttingDown: (data: ActionProcessor<any>) => {
```

Comment:
these unused params can probably be removed

Reply:
Yeah - this one was tricky! In the templates Actionhero generates, I went with generating these methods with all possible args... so now we have them :/

Reply:
I'll remove